keras/utils/vis_utils.py

```diff
@@ -13,7 +13,10 @@
 
 
 def _check_pydot():
-    if not (pydot and pydot.find_graphviz()):
+    try:
+        # Attempt to create an image of a blank graph to check the pydot/graphviz installation.
+        pydot.Dot.create(pydot.Dot())
+    except:
```

Comment:
There is not a specific Exception class in this situation. pydot raises a generic Exception ([pydot source](https://github.com/erocarrera/pydot/blob/master/pydot.py#L1881)).